src/utils.ts

```diff
@@ -82,8 +86,18 @@ export namespace Utils {
           value: webAuthDomain,
           source: account.accountId(),
         }),
-      )
-      .build();
+      );
+    if (memo) {
+      try {
+        transactionBuilder.addMemo(Memo.id(memo));
+      } catch {
+        throw new Error(
+          "invalid value for 'memo', must be 64-bit integer string",
+        );
```

Comment:
❓ What's the motivation for hiding the error within and creating a new error in this function? Lookin at the code (link below) the error returned is already descriptive and and appropriate. I think we could drop the try/catch here and simply let the validation bubble up.

https://github.com/stellar/js-stellar-base/blob/7731ceacea24e2c441711140935886a3382ce1e7/src/memo.js#L101-L102

If we do need the try/catch and our own error, I think it should be wrapped around the smallest surface area that can cause the relevant error, i.e. by only try/catch-ing around the `Memo.id(...)`. As far as I can tell it is `Memo.id(...)` that validates the ID and raises the error. Because the memo is input, it could also make sense to call `Memo.id(...)` early on in the function so as to exit early on invalid input.

Reply:
Yea I had the same though. But I think the error message returned from `Memo.id()` doesn't provide the caller of the SEP-10 utility function with enough context. 

> `Expects a int64 as a string. Got ${value}`

From the SDK user's perspective, they don't know what 'Expects' the int64 unless they look at the stack trace.

I'm not too opinionated on this though, and regardless I'll updated the try/catch to only contain the `Memo.id()` call.

Reply:
I think we could improve the `Memo.id` and change it to read:
>Memo ID expects a int64 as a string, got ${value}.

Wdyt?

Reply:
That works for me 👍

Reply:
This message will need to be altered in the js-stellar-base project so for now I'll adjust the test to assert an error is thrown in general.